cluster/wnode/docker-compose.yml

```diff
@@ -0,0 +1,29 @@
+version: "3"
+services:
+  bootnode:
+    image: ethereum/client-go:alltools-latest
+    entrypoint: bootnode
+    command: -addr=:30303 -nodekey /static/keys/bootnode.key
+    volumes:
+      - ./../../static/keys:/static/keys:ro
+    networks:
+      cluster:
+        ipv4_address: 172.16.238.10
```

Comment:
Is it possible to use docker [network alias](https://docs.docker.com/compose/compose-file/#aliases) here instead of an IP address?

Reply:
Unfortunately, geth bootnodes and peers can only use IP addresses. There is no DNS resolution implemented, probably because there is no good decentralized alternative. Even if there was, bootnodes still would need to be specified with IP addresses because decentralized DNS would probably be stored on-chain so one would need to sync the blockchain before it could be used.

I tried to add code to our wnode-status binary that resolves hostnames before running a node and it worked but finally decided to use static IPs. It will be a problem if we decide to run it as a cluster in Docker Swarm but then we could split it into two steps: (1) start a bootnode, (2) use [docker-compose env support](https://docs.docker.com/compose/environment-variables/#the-env_file-configuration-option) to specify bootnodes and start Whisper nodes.

Reply:
Thanks a lot for a comprehensive answer! 👍 
I suspected it was something specific to the implementation details.